cmd/configure/configure.go

```diff
@@ -0,0 +1,103 @@
+package configure
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/dream11/odin/api/configuration"
+	"github.com/dream11/odin/app"
+	"github.com/dream11/odin/cmd"
+	"github.com/dream11/odin/internal/service"
+	viperConfig "github.com/dream11/odin/pkg/config"
+	"github.com/dream11/odin/pkg/dir"
+	auth "github.com/dream11/odin/proto/gen/go/dream11/od/auth/v1"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var odinAccessKey string
+var odinSecretAccessKey string
+var odinBackendAddress string
+var odinInsecure bool
+
+// default backend address
+const defaultBackendAddress = "odin-backend.d11dev.com:443"
+
+var configureClient = service.Configure{}
+var configureCmd = &cobra.Command{
+	Use:   "configure",
+	Short: "Configure odin",
+	Long:  "Configure odin using odin access key and odin secret access key",
+	Run: func(cmd *cobra.Command, args []string) {
+		execute(cmd)
+	},
+}
+
+func init() {
+	configureCmd.Flags().StringVar(&odinAccessKey, "access-key", "", "odin access key")
+	configureCmd.Flags().StringVar(&odinSecretAccessKey, "secret-access-key", "", "odin secret access key")
+	configureCmd.Flags().StringVar(&odinBackendAddress, "backend-address", "", "odin backend address with port")
+	configureCmd.Flags().BoolVarP(&odinInsecure, "insecure", "I", false, "odin insecure")
+	cmd.RootCmd.AddCommand(configureCmd)
+}
+
+func execute(cmd *cobra.Command) {
+	dirPath:=path.Join(os.Getenv("HOME"), "."+app.App.Name)
+	if err := dir.CreateDirIfNotExist(dirPath); err != nil {
+		log.Fatalf("Error creating the .%s folder: %v", app.App.Name, err)
+	}
+	configPath := path.Join(dirPath, "config")
+	if err := dir.CreateFileIfNotExist(configPath); err != nil {
+		log.Fatal("Error creating the config file")
+	}
+
+	config := viperConfig.GetConfig()
+	profile := viper.GetString("profile")
+
+	config.BackendAddress = getConfigKey("backend-address", odinBackendAddress, "ODIN_BACKEND_ADDRESS", config.BackendAddress, defaultBackendAddress)
+	config.Insecure = odinInsecure
+	config.Keys.AccessKey = getConfigKey("access-key", odinAccessKey, "ODIN_ACCESS_KEY", config.Keys.AccessKey, "")
+	config.Keys.SecretAccessKey = getConfigKey("secret-access-key", odinSecretAccessKey, "ODIN_SECRET_ACCESS_KEY", config.Keys.SecretAccessKey, "")
+	setConfig(profile, config)
```

Comment:
Remove this as the config is written later

Reply:
Done